### PR TITLE
fixes

### PR DESCRIPTION
**Pull request by austinhallock**

### CODE REVIEW

File: src/components/friends/index.coffee [hunk 1]
```diff
@@ -19,7 +22,16 @@ module.exports = class Friends
       $groupIcon: new Icon()
       $addIcon: new Icon()
       $spinner: new Spinner()
-      friends: z.observe User.getFriends()
+      friends: z.observe User.getFriends().then (friends) ->
+        Promise.map friends, (friend) ->
```
Comment (Zolmeister) [inline src/components/friends/index.coffee]: Obviously wrong.

Comment (Zolmeister) [inline src/components/friends/index.coffee]: Well, ok not obviously, but `getFriends()` should fetch the friend

Comment (Zolmeister) [inline src/components/friends/index.coffee]: nvm, total garbage. ignore what I just said

File: src/components/friends/index.coffee [hunk 2]
```diff
@@ -19,7 +22,16 @@ module.exports = class Friends
...
+          recentGamesHref = friend.links.recentGames?.href
```
Comment (Zolmeister) [inline src/components/friends/index.coffee]: `User.getRecentGames()`

Comment (Zolmeister) [inline src/components/friends/index.coffee]: nvm

File: src/root.coffee [hunk 3]
```diff
@@ -145,6 +145,7 @@ else
   z.router.add '/join', JoinPage
   z.router.add '/join/:fromUserId', JoinPage
   z.router.add '/login', LoginPage
+  z.router.add '/login/:fromUserId', LoginPage
```
Comment (Zolmeister) [inline src/root.coffee]: This should be a query param
I said this before, why wasn't it changed?

File: src/components/friends/index.coffee [hunk 4]
```diff
@@ -19,7 +22,16 @@ module.exports = class Friends
       $groupIcon: new Icon()
       $addIcon: new Icon()
       $spinner: new Spinner()
-      friends: z.observe User.getFriends()
+      friends: z.observe User.getFriends().then (friends) ->
```
Comment (Zolmeister) [inline src/components/friends/index.coffee]: This code bothers me. It really ought to be `User.getFriends({embed: 'recentGames'})`, but not worth changing now

**Comment by Zolmeister** [comment]
done

File: src/root.coffee (hunk 3 above)
Comment (austinhallock) [inline src/root.coffee]: didn't see your response on how to implement. found it now

**Comment by austinhallock** [comment]
https://github.com/claydotio/clay-mobile/commit/43fce968631a0fe21c6d348084390966b4881e0d

**Comment by Zolmeister** [comment]
conflicts

File: src/components/invite_landing/personal.coffee [hunk 5]
```diff
@@ -51,12 +51,21 @@ module.exports = class InviteLanding
               if isLoggedIn
                 User.addFriend(fromUser.id).then ->
                   z.router.go '/'
+                .catch (err) ->
+                  log.trace err
+                  z.router.go '/'
               else
-                z.router.go "/join/#{fromUser.id}"
+                z.router.go "/join?fromUser=#{fromUser.id}"
```
Comment (Zolmeister) [inline src/components/invite_landing/personal.coffee]: misspelled

File: src/pages/login/index.coffee [hunk 6]
```diff
@@ -9,16 +9,19 @@ styles = require './index.styl'
 localStyleConfig = require './vars.json'
 
 module.exports = class LoginPage
-  constructor: ->
+  constructor: ({}, {fromUserId}) ->
```
Comment (Zolmeister) [inline src/pages/login/index.coffee]: rename to just 'from'

**Comment by austinhallock** [comment]
https://github.com/claydotio/clay-mobile/commit/5b62b69220662e72c5fd45cbff75a02adc45f607

**Comment by Zolmeister** [comment]
rebase

**Comment by austinhallock** [comment]
done

**Pull request merged**